BlueprintUI/Sources/Element/ElementState.swift

```diff
@@ -140,10 +167,29 @@ final class ElementState {
         precondition(self.identifier == identifier)
         precondition(type(of: newElement) == type(of: element))
 
-        let isEquivalent = Self.elementsEquivalent(element, newElement)
+        let isEquivalent: Bool
+
+        switch comparability {
+        case .notComparable:
+            isEquivalent = false
+
+        case .comparableElement:
+            isEquivalent = Self.elementsEquivalent(element, newElement)
+
+        case .childOfComparableElement:
+            /// **Note**: We're equivalent always, because our parent
+            /// determines our equatability from its own state.
+            isEquivalent = true
+        }
 
         if isEquivalent == false {
             clearAllCachedData()
+
+            if comparability == .comparableElement {
+                recursiveForEach {
+                    $0.clearAllCachedDataIfNotComparable()
+                }
+            }
```

Comment:
> This might be expensive, do we need to recurse here?

We do need to, because children of comparable elements always cache measurements.

> Can we guarantee that we always compare parents before their children?

I believe so, since the tree iterates top down; since it's a tree, not a graph.

>  In what case will have a child with isEquivalent == true and a parent who is false?

Thinking about eg, a `Label` within a parent element that contains many `Labels`. Only one of them may have changed!